minari/data_collector/data_collector.py

```diff
@@ -520,6 +543,25 @@ def save_to_disk(
             os.path.join(self._tmp_dir.name, "tmp_dataset.hdf5"), "a", track_order=True
         )
 
+    # This function is designed the same way as `assert_infos_same_shape` in tests/common.py, but is a class function so has a `self` argument.
+    def check_infos_same_shape(self, info_1, info_2):
+        if len(info_1.keys()) != len(info_2.keys()):
+            return False
+        for key in info_1.keys():
+            if isinstance(info_1[key], dict):
+                if not self.check_infos_same_shape(info_1[key], info_2[key]):
+                    return False
+            elif isinstance(info_1[key], np.ndarray):
+                if not (info_1[key].shape == info_2[key].shape) and (
+                    info_1[key].dtype == info_2[key].dtype
+                ):
+                    return False
+            else:
+                raise ValueError(
+                    "Infos are in an unsupported format; see [Minari documentation](http://minari.farama.org/content/dataset_standards/) for supported formats."
+                )
+        return True
+
```

Comment:
at the very least, if sometimes the info is `{}` and sometimes it has a single key and value, there is no error and the infos list is shorter than the number of observations(very bad). I haven't tried for the setting where the same key has two different `np.ndarray` shapes at two different timesteps.

Reply:
@younik I don't understand what you mean by adding the check while step_data is added to the buffer. Isn't it already like that?

Also, I would make this function independent from the DataCollectorV0 class, move it to utils for example or just outside DataCollector. That way it can be used without creating an environment since this function can be used for other testing purposes.

Reply:
I added the checks during the addition of step_data to the buffer